src/containers/Dashboard.js

```diff
@@ -0,0 +1,163 @@
+import '@binaris/shift-code-transform/macro';
+import React from 'react';
+import { Chart } from 'primereact/chart';
+
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Container from 'react-bootstrap/Container';
+import Button from 'react-bootstrap/Button';
+import Alert from 'react-bootstrap/Alert';
+import Spinner from 'react-bootstrap/Spinner';
+
+import {
+  generateRemoteData,
+  initData,
+  getStoredData,
+} from '../../backend/dashboard-server';
+import { Display } from '../constants/enums';
+
+export default class Dashboard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chartData: {
+        labels: ['A', 'B', 'C'],
+        datasets: [
+          {
+            label: 'My First dataset',
+            data: [100, 200, 300],
+            backgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+            hoverBackgroundColor: ['#FF6384', '#13bdd8', '#FFCE56'],
+          },
+        ],
+      },
+      display: Display.FRONTEND,
+    };
+    this.generateLocalData = this.generateLocalData.bind(this);
+    this.getRemoteData = this.getRemoteData.bind(this);
+    this.fetchStoredData = this.fetchStoredData.bind(this);
+  }
+
+  componentDidMount() {
+    initData(this.state.chartData.datasets[0].data);
```

Comment:
Please handle the potential error here.
You might want to (not required) to start with loading state and only transition to frontend state after `initData` is complete similarly to `getRemoteData`.

Reply:
I don't think we should have a loader at this point, and I will explain: 
This function basically creates the document in the DB, we are showing this data only when the button was pressed (and then we are displaying the spinner/error respectively).
There is nothing that is loading and we are not waiting for something to happen.

I added try..catch to this block of code, in case our call will get Promise rejection. Currently, we won't support messaging/logs to show the user, so I ended up adding log to the console in case there is an error.